source/class/qx/event/handler/Keyboard.js

```diff
@@ -590,27 +590,38 @@ qx.Class.define("qx.event.handler.Keyboard", {
       var keyIdentifier;
 
       // Use: keyCode
+      let returnPromise = null;
+      var tracker = {};
+
       if (keyCode || (!keyCode && !charCode)) {
         keyIdentifier = qx.event.util.Keyboard.keyCodeToIdentifier(keyCode);
 
-        return this._fireSequenceEvent(domEvent, eventType, keyIdentifier);
+        returnPromise = qx.event.Utils.track(tracker, () =>
```

Comment:
Why did you change the return statements to returnPromise? IMHO the return statements are much more readable. And    you did not change it in loine 619. This is a little bit inconsistent.

Reply:
You're kind of right, it is better to declare and return `returnPromise` in the true branch of the if statement only. File updated.

However, because the `if (charcode)` statement may or may not execute (although it should execute 90% of the time), the `returnPromise` way is the only way to ensure that that promise only resolves when all the listeners have fully executed.